app/src/main/java/ru/henridellal/emerald/BackupPreference.java

```diff
@@ -19,33 +16,44 @@ public BackupPreference(Context c) {
 	public BackupPreference(Context c, AttributeSet attr) {
 		super(c, attr);
 	}
-	@Override
-	public void onClick() {
-		// request runtime permissions (Marshmallow+)
-        if (Build.VERSION.SDK_INT >= 23) {
-        	if ((getContext().checkSelfPermission(Manifest.permission.READ_EXTERNAL_STORAGE) == PackageManager.PERMISSION_DENIED)
-        		|| (getContext().checkSelfPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_DENIED)) {
-        		((Options)getContext()).requestPermissions(new String[]{Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE}, 0);
-        		return;
-        	} else {
-        		super.onClick();
-        	}
-        } else {
-        	super.onClick();
-        }
-	}
+
 	@Override
 	protected View onCreateDialogView() {
 		int mode = 0;
 		if ("backup".equals(getKey())) {
-				mode = 0;
+			mode = 0;
 		} else if ("restore".equals(getKey())) {
-				mode = 1;
-		} else if ("convert".equals(getKey())) {
```

Comment:
I removed this because I couldn't find any usage of this key

Reply:
I've used this mode to check the database content. It isn't needed anymore.